### DISCUSSION TITLE
Ssh Private Repo from RaspberryPi

### DISCUSSION THREAD

**Discussion by padbd-tx** [community/community#22730] [New to GitHub]
<p>So I can’t quite find the right solution, after hours searching and trying different solutions.<br>
I’m trying to set up ssh from my RaspberryPi to work with a private repo on my GitHub. I was using https fine, but was annoyed with having to enter my username and password every time I pushed a commit. After switching to ssh from https, now I get the error:</p>
<p><a href="mailto:pi@github.com">pi@github.com</a>: Permission denied (publickey).<br>
fatal: Could not read from remote repository.<br>
Please make sure you have the correct access rights<br>
and the repository exists.</p>
<p>What am I doing wrong?!</p>

**Comment by airtower-luna** [comment]
<p>Did you add the SSH key of the raspi to your Github account, as described <a href="https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account">here</a>? I’d skip the <code>xclip</code> stuff though and just <code>cat</code> the public key file to copy. :wink:</p>

**Comment by padbd-tx** [comment]
<p>Thank for the reply! I did add it to my GitHub account - several times, actually. Following the steps to generate a new key a few times. I did use xclip to copy it though, but it seemed to accept it. I do get a “Hi txtack! You’ve successfully authenticated…” message after running the <code>ssh -vT git@github.com</code> command, but still get the permission denied message when I try <code>git pull</code></p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="txtack" data-post="1" data-topic="126135">
<div class="title">
<div class="quote-controls"></div>
 txtack:</div>
<blockquote>
<p><a href="mailto:pi@github.com">pi@github.com</a>: Permission denied (publickey).</p>
</blockquote>
</aside>
<p>Wait a minute, that looks like git tries to login as the user <code>pi</code> instead of <code>git</code> as it should. Please check <code>git remote -v</code>, the URL for the Github repository needs to follow the pattern <code>git@github.com:USER/REPO.git</code>.</p>

**Comment by padbd-tx** [comment]
<p>OMG I THINK THAT’S IT! I didn’t click on “use SSH” when I copied the repo link from GitHub, because I initially cloned it using HTTPS! When I switched it on my pi, I only switched https to ssh! It sounds like that should fix it, I’m now trying to remember how I changed that ssh URL and I’ll be trying it!</p>
<p>(if you can’t tell, I’m new to this but having fun learning about it all)</p>

**Comment by padbd-tx** [comment]
<p>That was it!</p>
<p>Here’s what it was: <code>ssh://github.com/txtack/TackleBot.git</code><br>
Changed it to: <code>git@github.com:txtack/TackleBot.git</code><br>
Ran <code>git pull</code><br>
And it worked!!! Thank you so much!!!</p>

**Comment by airtower-luna** [comment]
<p>Nice, I’m glad to hear that! :smile:</p>
